src/CoffeeNation.Dependency/DependencyExtensions.cs

```diff
@@ -0,0 +1,79 @@
+﻿using System.Diagnostics.CodeAnalysis;
+using CoffeeNation.Core;
+using CoffeeNation.Core.Interfaces;
+using CoffeeNation.Data;
+using CoffeeNation.Data.Formatter;
+using CoffeeNation.Data.Interfaces;
+using CoffeeNation.Data.Interfaces.Formatter;
+using CoffeeNation.Data.Interfaces.Parser;
+using CoffeeNation.Data.Interfaces.Provider;
+using CoffeeNation.Data.Parser;
+using CoffeeNation.Data.Provider;
+using CoffeeNation.Repository;
+using CoffeeNation.Repository.Interfaces;
+using CoffeeNation.Service;
+using CoffeeNation.Service.Facade;
+using CoffeeNation.Service.Interfaces;
+using CoffeeNation.Service.Interfaces.Facade;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace CoffeeNation.Dependency
+{
+    [ExcludeFromCodeCoverage]
+    public static class DependencyExtensions
+    {
+        public static IServiceCollection ConfigureDependencies(this IServiceCollection services, string[] arguments)
+        {
+            services.AddCoreRegistrations();
+            services.AddServiceRegistrations();
+            services.AddRepositoryRegistrations();
+            services.AddDataRegistrations(arguments);
+
+            return services;
+        }
+
+        private static void AddCoreRegistrations(this IServiceCollection services)
+        {
+            services.AddTransient<IDistanceCalculator, CartesianDistanceCalculator>();
```

Comment:
In this case you are right, there is no real benefit in using a transient scope. Due to the stateless behaviour of the DistanceCalculator class, it could have been as well singleton. Will update code shortly.